.github/actions/check-ext-build/action.yml

```diff
@@ -1,27 +1,52 @@
 name: Check Ext build
 description: Checks if the build is a dependabot build or external contribution
 
+inputs:
+  gh-token:
+    description: 'GitHub token'
+    required: true
+
+outputs:
+  skip-build:
+    description: "Skip the build"
+    value: ${{ steps.check_build.outputs.SKIP_BUILD }}
+
+permissions:
+  contents: read
+  actions: read
+  checks: write
+
 runs:
   using: composite
   steps:
-
     - name: Check build
+      id: check_build
       env:
+        GH_TOKEN: ${{ inputs.gh-token }}
         SENDER_LOGIN: ${{ github.event.sender.login }}
         EVENT_ACTION: ${{ github.event.action }}
         LABEL: ${{ github.event.label.name }}
         PR_REPO_NAME: ${{ github.event.pull_request.head.repo.full_name }}
         REPO_NAME: ${{ github.repository }}
+        GITHUB_PR_NUMBER: ${{github.event.pull_request.number}}
```

Comment:
you will not have the PR number on push events: you need to retrieve it from the sha commit with rest api https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-commits

$ gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/activiti/activiti/commits/[sha]/pulls | jq -r '.[].number'